src/main/java/org/wso2/carbon/apimgt/securityenforcer/PingAISecurityHandler.java

```diff
@@ -0,0 +1,331 @@
+ * It will fetch some of meta data from incoming message and send them to PING API Security Enforcer.
+ */
+public class PingAISecurityHandler extends AbstractHandler {
+
+    private static final Log log = LogFactory.getLog(PingAISecurityHandler.class);
+
+    public PingAISecurityHandler() {
+        log.debug("Ping AI Security Handler initialized");
+    }
+
+    /**
+     * This method will handle the request. For every request gateway receives, this is method will invoke first for this handler
+     */
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+
+        long handleRequestStartTime = System.nanoTime();
+        String requestCorrelationID = SecurityUtils.getAndSetCorrelationID(messageContext);
+        try {
+            if (authenticate(messageContext, requestCorrelationID)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Handle Request Time for the request " + requestCorrelationID + " is " + (System.nanoTime()
+                                    - handleRequestStartTime) + " Nano seconds");
+                }
+                SecurityUtils.updateLatency(System.nanoTime() - handleRequestStartTime, messageContext);
+                return true;
+            }
+        } catch (AISecurityException e) {
+            if (log.isDebugEnabled()) {
+                long difference = System.nanoTime() - handleRequestStartTime;
+                String messageDetails = logMessageDetails(messageContext);
+                log.debug("Request " + requestCorrelationID + " failed. " + messageDetails + ", elapsedTimeInNano"
+                        + difference);
+            }
+            handleAuthFailure(messageContext, e);
+        } finally {
+            SecurityUtils.updateLatency(System.nanoTime() - handleRequestStartTime, messageContext);
+        }
+        return false;
+    }
+
+    /**
+     * This method will handle the response.
+     */
+    @Override
+    public boolean handleResponse(MessageContext messageContext) {
+        long handleResponseStartTime = System.nanoTime();
+        try {
+            sendResponseDetailsToASE(messageContext);
+            if (log.isDebugEnabled()) {
+                log.debug("Handle Response Time " + (System.nanoTime() - handleResponseStartTime));
+            }
+            SecurityUtils.updateLatency(System.nanoTime() - handleResponseStartTime, messageContext);
+            return true;
+
+        } catch (AISecurityException e) {
+            if (log.isDebugEnabled()) {
+                long difference = (System.nanoTime() - handleResponseStartTime);
+                String messageDetails = logMessageDetails(messageContext);
+                log.debug("Call to Ping ASE : " + messageDetails + ", elapsedTimeInNanoseconds=" + difference);
+            }
+            handleAuthFailure(messageContext, e);
+        }
+        return false;
```

Comment:
You can just have a single return true statement at the end in place of having separate return true and return false statements